components/callbacks/executors.go

```diff
@@ -144,17 +147,30 @@ func (e taskExecutor) loadInvocationArgs(
 			if err != nil {
 				return err
 			}
+
+			completion, err := target.GetNexusCompletion(ctx, callback.GetRequestId())
+			if err != nil {
+				return err
+			}
+
+			// CHASM internal callbacks make use of Nexus as their callback delivery
+			// mechanism, but with a special prefix.
+			if strings.HasPrefix(variant.Nexus.Url, chasm.NexusCompletionHandlerURL) {
+				chasmInvokable := chasmInvocation{}
+				chasmInvokable.nexus = variant.Nexus
+				chasmInvokable.attempt = callback.Attempt
+				invokable = chasmInvokable
+				break
+			}
+
 			// variant struct is immutable and ok to reference without copying
 			nexusInvokable := nexusInvocation{}
```

Comment:
Same here, would be nice to just inline the whole field assignment (unrelated to your change).